app/command_autocomplete.go

```diff
@@ -12,6 +14,11 @@ import (
 	"github.com/mattermost/mattermost-server/v5/model"
 )
 
+// DynamicListProvider dynamically provides auto-completion for built-in commands.
+type DynamicListProvider interface {
```

Comment:
Can we incorporate a sense of `Autocomplete` here? Possibly `AutocompleteDynamicArgProvider`?

Reply:
Sounds good.  Done via https://github.com/mattermost/mattermost-server/pull/16196/commits/ccb5167c30a853d6f9732858e780cb74a756b381